src/mono/mono/metadata/class-init.c

```diff
@@ -1039,18 +1039,14 @@ class_composite_fixup_cast_class (MonoClass *klass, gboolean for_ptr)
 	case MONO_TYPE_U2:
 		klass->cast_class = mono_defaults.int16_class;
 		break;
-	case MONO_TYPE_U4:
-#if TARGET_SIZEOF_VOID_P == 4
 	case MONO_TYPE_I:
```

Comment:
`MONO_TYPE_BOOLEAN` -> `byte_class` normalization above looks suspect as well. I do not think regular CoreCLR does normalization like that anywhere.

Reply:
@jkotas I think the `bool -> int8` fixup comes from when I misread I.8.7.2:

> A location type T is *compatible-with* a location type U if and only if one of the following holds.
> 1. T and U are not managed pointer types and T is *compatible-with* U according to the definition in §I.8.7.1.
> 2. T and U are both managed pointer types and T is *pointer-element-compatible-with* U.
> A managed pointer type T is *pointer-element-compatible-with* a managed pointer type U if and
> only if T has *verification type* V and U has *verification type* W and V is identical to W.

The *verification type* of a bool is defined in I.8.7 as

> 1. If the reduced type of T is:
>    a. int8 or bool, then its verification type is int8.


The problem is I think I applied the *verification type* rules to **un**managed pointer types (ie `bool*`) to managed pointer types (ie `bool&`)